simd/src/Kokkos_SIMD_Common_Math.hpp

```diff
@@ -45,7 +45,7 @@ KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION T
   auto const& v = x.impl_get_value();
   auto const& m = x.impl_get_mask();
   auto result   = Kokkos::reduction_identity<T>::min();
-  for (std::size_t i = 0; i < v.size(); ++i) {
+  for (auto i = 0; i < v.size(); ++i) {
```

Comment:
I am not sure `auto` is a good idea here.
It makes `i`  be an `int` so we type more and rely on size() returning a signed integer type.

Reply:
Converted them to `simd_size_t` so that they stay consistent with what `size()` will return.